modules/phase_field/src/action/GrandPotentialKernelAction.C

```diff
@@ -65,8 +66,20 @@ GrandPotentialKernelAction::validParams()
   parameters.addParam<bool>(
       "use_displaced_mesh", false, "Whether to use displaced mesh in the kernels");
   MultiMooseEnum anisotropy("true=1 false=0", "false");
+  parameters.addParam<bool>("mass_conservation", false, "Whether strict mass conservation formulation is used or not");
   parameters.addRequiredParam<MultiMooseEnum>(
       "anisotropic", anisotropy, "Whether or not each mobility is anisotropic");
+  parameters.addParam<std::vector<NonlinearVariableName>>(
+      "concentrations", "List of concentration variables for strict mass conservation");
+  parameters.addParam<std::vector<MaterialPropertyName>>(
+      "body_forces", "List of body forces coefficients for strict mass conservation formulation."
```

Comment:
Please changes names as described above to `h_c_min` or something along those lines and also the description